lib/push.js

```diff
@@ -230,63 +235,67 @@ module.exports = function (log, db) {
     sendPush: function sendPush(uid, devices, reason, options) {
       options = options || {}
       var events = reasonToEvents[reason]
-      return P.all(
-        devices.map(function (device) {
-          var deviceId = device.id.toString('hex')
+      // There's no spec-compliant way to error out as a result of having
+      // too many devices to notify.  For now, just log metrics about it.
+      if (devices.length > MAX_ACTIVE_DEVICES) {
+        reportPushError(new Error(ERR_TOO_MANY_DEVICES), uid)
+      }
+      return P.each(devices, function(device) {
```

Comment:
Unless I missed it, it doesn't look like there is a test that asserts we call `each` rather than `all` here. If `all` is the thing that monopolised all the resources, should we have one?